src/language-js/printer-estree.js

```diff
@@ -2054,6 +2055,111 @@ function printPathNoParens(path, options, print, args) {
       return join(literalline, n.value.raw.split(/\r?\n/g));
     case "TemplateLiteral": {
       const expressions = path.map(print, "expressions");
+      const parentNode = path.getParentNode();
+
+      /**
+       * describe.each`table`(name, fn)
+       * describe.only.each`table`(name, fn)
+       * describe.skip.each`table`(name, fn)
+       * test.each`table`(name, fn)
+       * test.only.each`table`(name, fn)
+       * test.skip.each`table`(name, fn)
+       *
+       * Ref: https://github.com/facebook/jest/pull/6102
+       */
+      const jestEachTriggerRegex = /^[xf]?(describe|it|test)$/;
+      if (
+        parentNode.type === "TaggedTemplateExpression" &&
+        parentNode.quasi === n &&
+        parentNode.tag.type === "MemberExpression" &&
+        parentNode.tag.property.type === "Identifier" &&
+        parentNode.tag.property.name === "each" &&
+        ((parentNode.tag.object.type === "Identifier" &&
+          jestEachTriggerRegex.test(parentNode.tag.object.name)) ||
+          (parentNode.tag.object.type === "MemberExpression" &&
+            parentNode.tag.object.property.type === "Identifier" &&
+            (parentNode.tag.object.property.name === "only" ||
+              parentNode.tag.object.property.name === "skip") &&
+            parentNode.tag.object.object.type === "Identifier" &&
+            jestEachTriggerRegex.test(parentNode.tag.object.object.name)))
+      ) {
+        /**
+         * a    | b    | expected
+         * ${1} | ${1} | ${2}
+         * ${1} | ${2} | ${3}
+         * ${2} | ${1} | ${3}
+         */
+        const headerColumnNames = n.quasis[0].value.raw
+          .trim()
+          .split(/\s*\|\s*/);
+        if (
+          headerColumnNames.length > 1 ||
+          headerColumnNames.some(column => column.length !== 0)
+        ) {
+          const oneLineExpressions = expressions
+            .map(removeLineBreaks)
```

Comment:
This is dangerous, i’m sure that we can craft examples where it would generate invalid code with comments and asi

Reply:
What I would do instead is to print it and if there happen to be a line break then to ignore this line from the width computation